### PR TITLE
add jit mode for ipex to improve inference performance for pipeline

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sywangyi** [comment]
@echarlaix @yao-matrix @jianan-gu please help review

**Comment by sywangyi** [comment]
@mfuntowicz

**Comment by sywangyi** [comment]
@juliensimon

**Comment by sywangyi** [comment]
@kding1

**Comment by juliensimon** [comment]
@mfuntowicz @echarlaix Can you please take a look? This seems to have a nice impact on inference performance, and if so, I'd like to use it for the upcoming inference post. Thank you.

File: optimum/intel/ipex/inference.py
Comment (echarlaix) [inline optimum/intel/ipex/inference.py:99]: Dummy inputs can also be generated using the optimum.exporters appropriated [configuration](https://github.com/huggingface/optimum/blob/v1.6.1/optimum/exporters/onnx/__main__.py#L77-#L78) 
```
dummy_inputs = config.generate_dummy_inputs(framework="pt")
```
It will be extended in the future but would would require us to add `onnx` as a new dependecy until then

**Review by echarlaix — COMMENTED** [review]
Thanks for this addition @sywangyi ! I'll let @mfuntowicz approve this PR before merging

**Comment by mfuntowicz** [comment]
The JIT property seems to enable some additional speedup for some models but I'm not sure it would be compatible with all our usages.

For instance, when working with text generation, the optional nature of `past_keys_values` would introduce some burden to make it right.

Still, I think it's an interesting feature, would it be possible to move the ability to JIT a model to the `inference_mode` constructor to let the user decide? Also, jitting might not be the default path for PyTorch users along with introducing some "compilation" time for every new shape provided.

If we include this as a customization of `inference_mode` I would be ok to merge 👍🏻

**Review by mfuntowicz — APPROVED** [review]
LGTM! 👍🏻

**Comment by juliensimon** [comment]
Thanks, I'll give it a go.

**Comment by juliensimon** [comment]
Happy to confirm a spectacular improvement. Latency is down another 20-25%. Well done.

**Pull request merged**
